File: src/main/java/br/com/estimular/atendimento/service/messaging/producer/TranscreverProducer.java
package br.com.estimular.atendimento.service.messaging.producer;

import org.apache.kafka.common.serialization.ByteArraySerializer;
import org.apache.kafka.common.serialization.StringSerializer;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class TranscreverProducer extends AbstractProducer<String, byte[]> {

    public TranscreverProducer(@Value("${kafka.url}") String bootstrapServer,
                               @Value("${kafka.topico.transcrever}") String topico) {
        super(bootstrapServer, topico, StringSerializer.class, ByteArraySerializer.class);
    }
}

File: src/main/kotlin/com/clinica/atendimento/service/producer/TranscreverProducer.kt
package com.clinica.atendimento.service.producer

import org.apache.kafka.common.serialization.ByteArraySerializer
import org.apache.kafka.common.serialization.StringSerializer
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class TranscreverProducer(
    @Value("\${kafka.url}") bootstrapServer: String,
    @Value("\${kafka.topico.transcrever}") topico: String
) : AbstractProducer<String, ByteArray>(
    bootstrapServer,
    topico,
    StringSerializer::class.java,
    ByteArraySerializer::class.java
)